src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/PrivateRepositories.java

```diff
@@ -27,20 +27,20 @@ public PrivateRepository find(String username, String repositoryTitle) {
 		Preconditions.checkNotNull(username);
 		Preconditions.checkNotNull(repositoryTitle);
 
-		return ensureNotNull(query().from(privateRepository)
+		return ensureNotNull(query().selectFrom(privateRepository)
 				.where(privateRepository.owner.netId.equalsIgnoreCase(username)
 					.and(privateRepository.title.equalsIgnoreCase(repositoryTitle)))
-				.singleResult(privateRepository),
+				.fetchOne(),
```

Comment:
If there is no result it still returns null, leaving that logic valid, however it is stricter in the sense that for this query would there be two values, it throws a Not-unique exception.